lib/include/openamp/rpmsg_virtio.h

```diff
@@ -75,8 +75,26 @@ struct rpmsg_virtio_device {
 	struct rpmsg_virtio_shm_pool *shpool;
 };
 
-#define RPMSG_REMOTE	VIRTIO_DEV_SLAVE
-#define RPMSG_MASTER	VIRTIO_DEV_MASTER
+#define RPMSG_REMOTE	VIRTIO_DEV_DEVICE
+#define RPMSG_HOST	VIRTIO_DEV_DRIVER
+
+#ifdef VIRTIO_DEV_MASTER
+#if defined(__GNUC__)
+_Pragma("GCC warning\"Please use VIRTIO_DEV_DRIVER instead of VIRTIO_DEV_MASTER which is deprecated\"")
+#elif defined(__ICCARM__)
+#define DO_PRAGMA(x) _Pragma(#x)
+DO_PRAGMA(message("Warning: Please use VIRTIO_DEV_DRIVER instead of VIRTIO_DEV_MASTER which is deprecated"))
+#endif
```

Comment:
```
#if defined(__GNUC__)
_Pragma("GCC warning\" deprecated VIRTIO_MASTER_ONLY is replaced by VIRTIO_DRIVER_ONLY\"")
#elif defined(__ICCARM__)
#define DO_PRAGMA(x) _Pragma(#x)
DO_PRAGMA(message("Warning: deprecated VIRTIO_MASTER_ONLY is replaced by VIRTIO_DRIVER_ONLY"))
#endif

```
should avoid too long line ans seems aligned with the cmake comments

Reply:
I made more tests with Gcc Armcc and arm 

> ```
> #if defined(__GNUC__)
> _Pragma("GCC warning\" deprecated VIRTIO_MASTER_ONLY is replaced by VIRTIO_DRIVER_ONLY\"")
> #elif defined(__ICCARM__)
> #define DO_PRAGMA(x) _Pragma(#x)
> DO_PRAGMA(message("Warning: deprecated VIRTIO_MASTER_ONLY is replaced by VIRTIO_DRIVER_ONLY"))
> #endif
> ```
> 
> should avoid too long line ans seems aligned with the cmake comments

In fact, "_Pragma" doesn't work for armcc. Why not make it simpler by using #warning, which should work for all compilers?
```
#ifdef VIRTIO_MASTER_ONLY
#define VIRTIO_DRIVER_ONLY
#warning("Please use VIRTIO_DRIVER_ONLY instead of VIRTIO_MASTER_ONLY which is deprecated\")
#endif

#ifdef VIRTIO_SLAVE_ONLY
#define VIRTIO_DEVICE_ONLY
#warning("Please use VIRTIO_DEVICE_ONLY instead of VIRTIO_SLAVE_ONLY which is deprecated\")
#endif
```

Reply:
That looks much better. I will switch to \#warning.